File: Infrastructure/Logic/IBookDM.cs
using System;
using ViewModels;

namespace Infrastructure.Logic
{
    public interface IBookDM : IDisposable
    {
        void Delete(long id);
        void Update(BookVM model);
        void Create(BookVM model);
        BookVM Get(long? id);
        DataTableResponseVM<BookVM> GetList(DataTableRequestVM model);
    }
}

File: InfrastructureVB/Logic/IBookDM.vb
Imports ViewModels

Namespace Logic
    Public Interface IBookDM
        Inherits IDisposable

        Sub Update(model As BookVM)
        Sub Create(model As BookVM)
        Function [Get](model As Long?) As BookVM
        Sub Delete(id As Long)

        Function GetList(model As DataTableRequestVM) As DataTableResponseVM(Of BookVM)

    End Interface
End Namespace